csrc/python_frontend/fusion_record.h

```diff
@@ -582,84 +582,66 @@ struct SqueezeOpRecord : RecordFunctor {
       std::vector<int64_t> dims)
       : RecordFunctor(
             std::move(_args),
             std::move(_outputs),
             "ops.squeeze",
             serde::RecordType::SqueezeOp),
-        original_shape_(std::move(original_shape)),
         dims_(std::move(dims)) {}
   ~SqueezeOpRecord() override = default;
   RecordFunctor* clone() final {
     return new SqueezeOpRecord(*this);
   }
 
   //! Child specific hash function in lower 32 bits.
-  //! | 31 -------------- 16 | 15 --------------  0 |
-  //! | Squeeze Dim hash     | original_shape hash  |
+  //! | 31 -------------------------------------  0 |
+  //! | Squeeze Dim hash                            |
   size_t hash() const final {
     auto result = RecordFunctor::hash();
-    size_t original_shape_hash = 0;
-    for (auto shape : original_shape_) {
-      original_shape_hash ^= static_cast<size_t>(shape);
-    }
     size_t squeeze_dims_hash = 0;
     for (auto dim : dims_) {
       squeeze_dims_hash ^= static_cast<size_t>(dim);
     }
-    squeeze_dims_hash = (squeeze_dims_hash & 0xffff) << 16;
-    return result | squeeze_dims_hash | (original_shape_hash & 0xffff);
+    result = result | (squeeze_dims_hash & 0xffffffff);
+    return result;
   }
 
   bool operator==(const RecordFunctor& other) const final {
     auto result = false;
     if (auto child_ptr = dynamic_cast<const SqueezeOpRecord*>(&other)) {
-      result = RecordFunctor::operator==(other);
-      if (result) {
-        result = (original_shape_.size() == child_ptr->original_shape_.size());
-        if (result) {
-          for (size_t i = 0; i < dims_.size(); ++i) {
-            if (dims_[i] != child_ptr->dims_[i]) {
-              result = false;
-              break;
-            }
-          }
-        }
-        if (result) {
-          for (size_t i = 0; i < original_shape_.size(); ++i) {
-            if (original_shape_[i] != child_ptr->original_shape_[i]) {
-              result = false;
-              break;
-            }
-          }
-        }
-      }
+      result = RecordFunctor::operator==(other) && (dims_ == child_ptr->dims_);
     }
     return result;
   }
 
   void operator()(FusionState& fd) final {
     auto arg = fd.getFusionState(args_.at(0).index)->template as<TensorView>();
-    auto output = squeeze(arg, original_shape_, dims_);
+    const auto inp_dom =
+        TensorDomain::noReductions(arg->getMaybeRFactorDomain());
+    const auto ndims = inp_dom.size();
+    std::vector<bool> squeeze_dims(ndims, false);
+    for (auto dim : dims_) {
+      // Handle negative relative to the end dimensions specifications
+      if (dim < 0) {
+        dim = static_cast<int64_t>(squeeze_dims.size()) + dim;
+      }
+      NVF_CHECK(
+          (dim >= 0) && (static_cast<size_t>(dim) < squeeze_dims.size()),
+          "Squeeze dim is outside of Tensor size! Tensor Size: ",
+          squeeze_dims.size(),
+          " Dim: ",
+          dim);
+      squeeze_dims[dim] = true;
```

Comment:
I think we need to skip `squeeze_dims[dim]` if `inp_dom[dim]` is not a size-1 dimension.

Reply:
We cannot skip dimensions of a specific size from the python API as we are only constructing the fusion object.  We do not have the runtime sizes.

Reply:
This is the `operator()`. So at this phase, we at least have access to its TensorDomain.

i.e. we can check for `isBroadcast` and `isExpanded` to figure out if we can skip the squeeze of a certain domain.
Without doing this, we'll run into asserts later, since our cpp API `squeeze(TensorView*, const std::vector<bool>&)` asserts on that.

Alternatively, we can update the `squeeze` API and have it clean up the squeeze dimension there.

Reply:
I agree, we should only have a `SqueezeOp` node if `inp_dom[dim]->isBroadcast() && inp_dom[dim]->hasExpandedExtent()`. But maybe that is best placed in the `squeeze()` C++ ops?